### ISSUE TITLE
ExcelApi 1.1 not available in Office Pro Plus 2016

### ISSUE THREAD

**Issue by ndreisg** [OfficeDev/office-js#136]
ExcelApi 1.1 is not available in Office Professional Plus 2016 for one of my customers (one-time purchase installed via MSI).

I asked him to send me his Office Version and he sent me the following screenshot:

![screenshotaboutexcel](https://user-images.githubusercontent.com/6780383/40106311-bb0093f0-58f5-11e8-8718-265c024b2e3b.png)

According to your documentation, this version should support ExcelApi 1.1.
Regardless, my customer is facing the following error: 
`Object doesn't support property or method 'getResizedRange'`

## Expected Behavior
ExcelApi 1.1 should be available.

## Current Behavior
ExcelApi 1.1 is not available.

## Steps to Reproduce, or Live Example
I was not able to reproduce this issue on my machine, though I also have Office 2016 installed via MSI.

## Your Environment
Windows 10 
Office Professional Plus 2016 (one-time purchase installed via MSI)

## Useful Links
Related Issue on StackOverflow:
https://stackoverflow.com/questions/42563791/office-js-excel-host-api-no-longer-available-under-excel-2016-mso

**Comment by JuaneloJuanelo** [comment]
No repro. Please advise customer to update or reinstall MSI.

**Comment by JuaneloJuanelo** [comment]
Assinging to @Zlatkovsky (as per the discussion in stack)

**Comment by ndreisg** [comment]
As I mentioned above (and on SO) I also have Office installed via MSI.
But my version differs from the customers version (16.0.4639.1000).

I already advised the customer to install the latest updates, but he stated, that he has already installed all Windows Updates.

**Comment by Zlatkovsky** [comment]
@ndreisg , it looks like there were 5 APIs -- `getColumnsBefore`, `getColumnsAfter`, `getResizedRange`, `getRowsAbove` and `getRowsBelow` -- that were incorrectly marked in the documentation as 1.1, even though they technically are not.

In fact, if you look at the DefinitelyTyped d.ts file for them (or the IntelliSense you get in VS, that comes from this source or from the "-vsdoc" file in VS 2015 and earlier), you will see those listed as 1.2

https://raw.githubusercontent.com/DefinitelyTyped/DefinitelyTyped/master/types/office-js/index.d.ts

I've gone ahead and made the change to the documentation, it should be reflected in the live site in a day or so.  But here is the commit:  https://github.com/OfficeDev/office-js-docs/commit/cc5c70b9d5b7956c81cba9742fd78005eebe4c45

As for why it might work on one computer but not another:  the original Office 2016 RTM loads the file https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.00.debug.js, whereas the O365 subscription version loads https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.debug.js.  (I'm using "debug" versions here, where you can see the actual code in non-minified form; in practice, if you're referencing "office.js" rather than "office.debug.js", you would correspondingly not get the other debug files).  There was an update to the MSI to load the 16.01 files, but I'm not sure what exact update this went out on, and how many people actually got it.

With most of our APIs, the APIs require C++ code to work.  These 5 are actually an exception, since they can be "polyfilled" in JS alone, using other existing APIs as building blocks.  As such, they are somewhere between 1.1 and 1.2, even though the actual C++ native implementations didn't come until 1.3.

You can see this in the debug file.  Search for "getResizedRange", for example, and you'll see:

		Range.prototype.getResizedRange=function (deltaRows, deltaColumns) {
			if (!isExcel1_3OrAbove()) {
				this._ensureInteger(deltaRows, "getResizedRange");
				this._ensureInteger(deltaColumns, "getResizedRange");
				var referenceRange=(deltaRows >=0 && deltaColumns >=0) ? this : this.getCell(0, 0);
				return referenceRange.getBoundingRect(this.getLastCell().getOffsetRange(deltaRows, deltaColumns));
			}
			_throwIfApiNotSupported("Range.getResizedRange", _defaultApiSetName, "1.3", _hostName);
			return new Excel.Range(this.context, _createMethodObjectPath(this.context, this, "GetResizedRange", 1, [deltaRows, deltaColumns], false, true, null, 4));
		};

So, in summary:

1. The APIs were incorrectly marked as 1.1 even though they aren't; I've moved them to 1.2 instead in the docs, and that's what the IntelliSense always had as well.
2. If you want, you can "polyfill" them yourself, using the debug file as a way to see how exactly we did it.
3. There is a chance we can have even the regular MSI builds point to the 16.01 files even without updates required.  This won't get you any new C++-requiring APIs, but it will add these polyfilled APIs, as well as some JS-based fixes and improvements.  I will explore that option, but no guarantees.  Tracking it under https://github.com/OfficeDev/office-js/issues/138

**Comment by ndreisg** [comment]
Thanks for your investigations.

Could this be the update you mentioned?
https://support.microsoft.com/en-us/help/4018369/may-1-2018-update-for-office-2016-kb4018369 

On Stackoverflow (https://stackoverflow.com/questions/41023863/runtime-error-officeextension-promise-is-not-defined/41024098) you stated, that OfficeExtension.Promise is available on all hosts with API Level 1.1 since may. 
Would this also be true for hosts that use the 16.00 files?

**Comment by sameera** [comment]
@Zlatkovsky 
Could you please take a look at https://github.com/OfficeDev/office-js/issues/44 as well, which is also specific to ProPlus (build 16.0.4266.1003)
